Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/Space.java

```diff
@@ -0,0 +1,179 @@
+ */
+
+package org.kitodo.dataaccess.format.xml;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Enum implementing the XML white space handling value constants.
+ *
+ * @see "http://www.w3.org/TR/xml/#sec-white-space"
+ */
+enum Space {
+
+    /**
+     * Signals that white-space processing is acceptable for this element.
+     */
+    DEFAULT("default") {
+        @Override
+        /**
+         * Trimming white space the XML way.
+         */
+        public String trim(String s) {
+            State state = State.INITIAL;
+            StringBuilder result = new StringBuilder(s.length());
+            for (int i = 0; i < s.length(); i++) {
+                int codePoint = s.codePointAt(i);
+                if (!WHITE_SPACE.contains(codePoint)) {
+                    if (state.equals(State.WHITE_SPACE)) {
+                        result.append(' ');
+                    }
+                    result.appendCodePoint(codePoint);
+                    state = State.TEXT;
+                } else if (state.equals(State.TEXT)) {
+                    state = State.WHITE_SPACE;
+                }
+            }
+            return result.toString();
+        }
+    },
+
+    /**
+     * Signals the intention that in that element the white space should be
+     * preserved.
+     */
+    PRESERVE("preserve") {
+        @Override
+        /**
+         * Trimming is doing nothing.
+         */
+        public String trim(String s) {
+            return s;
+        }
+    };
+
+    /**
+     * States for a little state machine used in trimming white space.
+     */
+    private enum State {
+        /**
+         * Initial state. Preceeding white space is skipped.
+         */
+        INITIAL,
+        /**
+         * Last character was text. If the next character is text, do not
+         * prepend a white space.
+         */
+        TEXT,
+        /**
+         * Last character was non-initial white space. If the next character is
+         * text, a white space will be prepended.
+         */
+        WHITE_SPACE
+    }
+
+    /**
+     * Reverse look-up map for the enum constants.
+     */
+    private static final Map<String, Space> reversed;
+
+    /**
+     * Characters considered as white space.
+     */
+    private static Set<Integer> WHITE_SPACE = new HashSet<>(
+            new HashSet<>(Arrays.asList((int) '\n', (int) '\r', (int) '\t', (int) ' ')));
+
+    static {
+        reversed = new HashMap<>();
+        for (final Space value : Space.values()) {
+            reversed.put(value.enumeratedValue, value);
+        }
+    }
+
+    /**
+     * Returns whether white space must be preserved to correctly represent this
+     * string in XML.
+     *
+     * @param s
+     *            string to examine
+     * @return if preservation of white space is necessary
+     */
+    public static boolean requiresPreservation(String s) {
```

Comment:
Short method parameter names should be avoided. Please use a longer and understandable name.